airbyte-webapp/src/components/connection/ConnectionForm/calculateInitialCatalog.ts

```diff
@@ -21,42 +21,49 @@ const clearBreakingFieldChanges = (nodeStream: SyncSchemaStream, breakingChanges
     return nodeStream;
   }
 
+  const { primaryKey, cursorField } = nodeStream.config;
+
   let clearPrimaryKey = false;
   let clearCursorField = false;
 
   for (const streamTransformation of breakingChangesByStream) {
-    // get all of the removed field paths for this transformation
-    const removedFieldPaths = streamTransformation.updateStream?.map((update) => update.fieldName);
-
-    if (!removedFieldPaths?.length) {
+    if (!streamTransformation.updateStream || !streamTransformation.updateStream?.length) {
       continue;
     }
 
+    // get all of the removed field paths for this transformation
+    const breakingFieldPaths = streamTransformation.updateStream
+      .filter(({ breaking }) => breaking)
+      .map((update) => update.fieldName);
+
     // if there is a primary key in the config, and any of its field paths were removed, we'll be clearing it
+    // The source-defined primary key already been updated by verifySourceDefinedProperties
```

Comment:
is it worth clarifying this by saying `"The source-defined primary key, if one exists..."`

Reply:
I will remove these. These were from an older solution that originally involved checking source-defined here, but with the check on breaking flag, it should do the right thing.